include/oneapi/tbb/detail/_flow_graph_impl.h

```diff
@@ -424,6 +441,11 @@ inline bool is_graph_active(graph& g) {
     return g.my_is_active;
 }
 
+inline bool is_thread_in_graph_arena(graph& g) {
+    __TBB_ASSERT(g.my_task_arena && g.my_task_arena->is_active(), nullptr);
+    return r1::execution_slot(*g.my_task_arena) != d1::slot_id(-1);
```

Comment:
Do we want to extend `task_arena` API with new method?

Reply:
This question needs to be discussed in details and the API and the applicability of this new method should also be worked out. Since it does not block the current PR, I am sure that it should be considered separately and in terms or separate PR.
We can open an issue to save this conversation for later discussions.

Reply:
I have opened an issue not to forget about it in the future. I guess the discussion can be closed for now